File: rlpark.plugin.examples/pysrc/CreateIntroSession.py
from rlpark.plugin.irobot.data import CreateAction;
from rlpark.plugin.irobot.data import IRobotDrops;
from rlpark.plugin.irobot.robots import CreateRobot;
from zephyr.plugin.core.api.synchronization import Clock
import zepy

if __name__ == '__main__':
    robot = CreateRobot("/dev/cu.ElementSerial-ElementSe")
    clock = Clock()
    zepy.advertise(robot, clock)
    bumpRightObsIndex = robot.legend().indexOf(IRobotDrops.BumpRight)
    bumpLeftObsIndex = robot.legend().indexOf(IRobotDrops.BumpLeft)
    robot.safeMode()
    while clock.tick():
      obs = robot.waitNewObs()
      wheelLeft = 150 if obs[bumpRightObsIndex] == 0 else -150
      wheelRight = 150  if obs[bumpLeftObsIndex] == 0 else -70
      robot.sendAction(CreateAction(wheelLeft, wheelRight))
      

File: rlpark.example.robots/src/rlpark/example/robots/irobot/CreateIntroSession.java
package rlpark.example.robots.irobot;

import rlpark.plugin.irobot.data.CreateAction;
import rlpark.plugin.irobot.data.IRobotDrops;
import rlpark.plugin.irobot.robots.CreateRobot;
import zephyr.plugin.core.api.Zephyr;
import zephyr.plugin.core.api.synchronization.Clock;

public class CreateIntroSession {
  public static void main(String[] args) {
    Clock clock = new Clock("Create");
    CreateRobot robot = new CreateRobot("/dev/cu.ElementSerial-ElementSe");
    Zephyr.advertise(clock, robot);
    int bumpRightObsIndex = robot.legend().indexOf(IRobotDrops.BumpRight);
    int bumpLeftObsIndex = robot.legend().indexOf(IRobotDrops.BumpLeft);
    robot.safeMode();
    while (!robot.isClosed() && clock.tick()) {
      double obs[] = robot.waitNewObs();
      double wheelLeft = obs[bumpRightObsIndex] == 0 ? 150 : -150;
      double wheelRight = obs[bumpLeftObsIndex] == 0 ? 150 : -70;
      robot.sendAction(new CreateAction(wheelLeft, wheelRight));
    }
  }
}
